scripts/deploy-simulation-alt.ts

```diff
@@ -179,58 +204,60 @@ async function main() {
   const poolsTicks: Record<string, BigNumber[]> = {};
+  const durations = [30 * 86400, 14 * 86400, 7 * 86400];
+  const rates = [FixedPoint.normalizeRate("0.10"), FixedPoint.normalizeRate("0.30"), FixedPoint.normalizeRate("0.50")];
 
   async function createCollectionPool(i: number) {
     const params = ethers.utils.defaultAbiCoder.encode(
       ["address", "address", "uint64[]", "uint64[]"],
-      [
-        collateralTokens[i],
-        wethTokenContract.address,
-        [30 * 86400, 14 * 86400, 7 * 86400],
-        [FixedPoint.normalizeRate("0.10"), FixedPoint.normalizeRate("0.30"), FixedPoint.normalizeRate("0.50")],
-      ]
+      [collateralTokens[i], wethTokenContract.address, durations, rates]
     );
     const createPoolTx = await poolFactory.createProxied(weightedRateCollectionPoolBeacon.address, params);
-    console.log("DEPLOYED COLLECTION POOl");
+    console.log("DEPLOYED COLLECTION POOL");
     return (await extractEvent(createPoolTx, poolFactory, "PoolCreated")).args.pool;
   }
 
   async function createRangedCollectionPool(i: number) {
     const params = ethers.utils.defaultAbiCoder.encode(
       ["address", "uint256", "uint256", "address", "uint64[]", "uint64[]"],
-      [
-        collateralTokens[i],
-        0,
-        2,
-        wethTokenContract.address,
-        [30 * 86400, 14 * 86400, 7 * 86400],
-        [FixedPoint.normalizeRate("0.10"), FixedPoint.normalizeRate("0.30"), FixedPoint.normalizeRate("0.50")],
-      ]
+      [collateralTokens[i], 0, 2, wethTokenContract.address, durations, rates]
     );
     const createPoolTx = await poolFactory.createProxied(weightedRateRangedCollectionPoolBeacon.address, params);
-    console.log("DEPLOYED RANGED POOl");
+    console.log("DEPLOYED RANGED POOL");
     return (await extractEvent(createPoolTx, poolFactory, "PoolCreated")).args.pool;
   }
 
   async function createSetCollectionPool(i: number) {
     const params = ethers.utils.defaultAbiCoder.encode(
       ["address", "uint256[]", "address", "uint64[]", "uint64[]"],
-      [
-        collateralTokens[i],
-        tokenIds,
-        wethTokenContract.address,
-        [30 * 86400, 14 * 86400, 7 * 86400],
-        [FixedPoint.normalizeRate("0.10"), FixedPoint.normalizeRate("0.30"), FixedPoint.normalizeRate("0.50")],
-      ]
+      [collateralTokens[i], tokenIds, wethTokenContract.address, durations, rates]
     );
     const createPoolTx = await poolFactory.createProxied(weightedRateSetCollectionPoolBeacon.address, params);
-    console.log("DEPLOYED SET POOl");
+    console.log("DEPLOYED SET POOL");
+    return (await extractEvent(createPoolTx, poolFactory, "PoolCreated")).args.pool;
+  }
+
+  async function createMerkleCollectionPool(i: number) {
+    const merkleTree = MerkleTree.buildTree(
+      tokenIds.map((tokenId) => [tokenId]),
+      ["uint256"]
+    );
+    const nodeCount = Math.ceil(Math.log2(tokenIds.length));
+    const metadataURI = "ipfs://QmZUvHgvb3p1souYvJtSMYS7MqjLao4PmXF4sxcTKe7m7G/";
```

Comment:
We should use `MerkleCollectionCollateralFilter` going forward for the type

Reply:
Updated the code 👍